File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/components/validators/MaxCharsValidator.java
package cz.cvut.fel.matyapav.afandroid.components.validators;

import android.view.View;
import android.widget.EditText;

import cz.cvut.fel.matyapav.afandroid.components.parts.ValidationRule;
import cz.cvut.fel.matyapav.afandroid.utils.Localization;

/**
 * Created by Pavel on 14.02.2016.
 */
public class MaxCharsValidator implements AFValidator {

    @Override
    public boolean validate(View field, StringBuilder errorMsgs, ValidationRule rule) {
        if(field instanceof EditText){
            EditText textfield = (EditText) field;
            if (textfield.getText() != null && textfield.getText().toString().length() > Integer.parseInt(rule.getValue())) {
                errorMsgs.append(Localization.translate("field.required", field.getContext()));
                return false;
            }
        }
        return true;
    }
}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/components/parts/validators/MaxCharsValidator.cs
using AFWindowsPhone.utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI.Xaml.Controls;

namespace AFWindowsPhone.builders.components.parts.validators
{
    class MaxCharsValidator : AFValidator
    {
        public bool validate(AFField field, StringBuilder errorMsgs, ValidationRule rule)
        {
            bool validationIsFine = true;
            if (Utils.isFieldWritable(field.getFieldInfo().getWidgetType()))
            {
                TextBox textfield = (TextBox)field.getFieldView();
                if (textfield.Text != null && textfield.Text.Length > Convert.ToInt32(rule.getValue()))
                {
                    validationIsFine = false;
                }
            }
            if (!validationIsFine)
            {
                errorMsgs.Append(Localization.translate("validation.maxchars"));
            }
            return true;
        }
    }
}
